scripts/mysql_system_tables_fix.sql

```diff
@@ -1415,8 +1415,30 @@ INSERT IGNORE INTO mysql.user VALUES ('localhost','mysql.pxc.sst.role','N','N','
 #  pxb-sst
 INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.pxc.sst.role', 'localhost', 'BACKUP_ADMIN', 'Y');
 INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.pxc.sst.role', 'localhost', 'INNODB_REDO_LOG_ARCHIVE', 'N');
-INSERT IGNORE INTO mysql.tables_priv VALUES ('localhost', 'PERCONA_SCHEMA', 'mysql.pxc.sst.role', 'xtrabackup_history', 'root\@localhost', CURRENT_TIMESTAMP, 'Alter,Select,Insert,Create', '');
-INSERT IGNORE INTO mysql.db VALUES ('localhost', 'PERCONA_SCHEMA', 'mysql.pxc.sst.role','N','N','N','N','Y','N','N','N','N','N','N','N','N','N','N','N','N','N','N');
+DELIMITER $$
+DROP PROCEDURE IF EXISTS conditional_grants $$
+CREATE PROCEDURE conditional_grants()
+BEGIN
+    DECLARE lctn INT DEFAULT 0;
+
+    -- Get the value of lower_case_table_names
+    SET lctn = @@lower_case_table_names;
+
+    IF lctn = 1 THEN
+        -- Case-insensitive system: use lowercase schema/table
+    INSERT IGNORE INTO mysql.tables_priv VALUES ('localhost', 'percona_schema', 'mysql.pxc.sst.role', 'xtrabackup_history', 'root\@localhost', CURRENT_TIMESTAMP, 'Alter,Select,Insert,Create', '');
```

Comment:
I think there is something wrong with indentation here and in following conditional branches.